modular_skyrat/modules/ashwalkers/code/buildings/railroad.dm

```diff
@@ -112,6 +113,22 @@
 	. = ..()
 	atom_storage?.show_contents(user)
 
+/obj/vehicle/ridden/rail_cart/attackby(obj/item/attacking_item, mob/user, params)
+	if(istype(attacking_item, /obj/item/stack/ore/glass))
+		var/obj/item/stack/ore/glass/use_item = attacking_item
+		if(GetComponent(/datum/component/simple_farm) || !use_item.use(10))
+			return ..()
+		AddComponent(/datum/component/simple_farm, TRUE, TRUE, list(0, 16))
+		RemoveElement(/datum/element/ridable)
+		return
+
+	if(attacking_item.tool_behaviour == TOOL_SHOVEL)
+		var/datum/component/remove_component = GetComponent(/datum/component/simple_farm)
+		if(remove_component)
+			qdel(remove_component)
+		AddElement(/datum/element/ridable, /datum/component/riding/vehicle/rail_cart)
```

Comment:
You might want to have a variable that stores whether it's got sand in it or not, otherwise you might end up adding the `ridable` element multiple times, and it would probably be faster than a `GetComponent()` check in that first if section.